soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x6

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022 William MARTIN <william.martin@power-lan.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32F030X6
+
+config SOC
+	default "stm32f030x6"
+
+config NUM_IRQS
+	default 29
```

Comment:
Could you please also check the soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f030x**4** : 
it is said to be like stm32f030x6 however the last irq entry is 27 for USART1_IRQn in the modules/hal/stm32/stm32cube/stm32f0xx/soc/tm32f030x6.h (assuming NUM_IRQS is 28)

Reply:
Yes, your right !

It's look like they have another errors in the stm32f030x4 dts (including `zephyr/dts/arm/st/f0/stm32f0.dtsi`)
- stm32f030x4 and stm32f030x6 have only one I2C controller.
- stm32f030x4 and stm32f030x6 do not have the timer 15
- stm32f030x4 and stm32f030x6 do not have the usart2

Do you prefers delete nodes from stm32f030x4 and stm32f030x6 dts, or update stm32f0.dtsi ?
The second solution have a large impact on others DTS file of the F0 family (F3x, F7x F9x).

https://www.st.com/resource/en/datasheet/stm32f030f4.pdf

Reply:
> Do you prefers delete nodes from stm32f030x4 and stm32f030x6 dts, or update stm32f0.dtsi ?

In this case, deleting nodes would be fine.